custom_imports: [projects.MTR.deploys]

input_names:
  [
    "obj_trajs",
    "obj_trajs_mask",
    "map_polylines",
    "map_polylines_mask",
    "map_polylines_center",
    "obj_trajs_last_pos",
    "track_index_to_predict",
    "intention_points",
  ]
output_names: ["pred_scores", "pred_trajs"]
input_shapes:
  "obj_trajs":
    value: ones
    shape: [1, 2, 11, 29] # [B, N, 11, 29]
    dtype: float
  "obj_trajs_mask":
    value: ones
    shape: [1, 2, 11] # [B, N, 11]
    dtype: bool
  "map_polylines":
    value: ones
    shape: [1, 768, 20, 9] # [B, 768, 20, 9]
    dtype: float
  "map_polylines_mask": # [B, 768, 20]
    value: ones
    shape: [1, 768, 20]
    dtype: bool
  "map_polylines_center": # [B, 768, 3]
    value: ones
    shape: [1, 768, 3]
    dtype: float
  "obj_trajs_last_pos": # [B, N, 3]
    value: ones
    shape: [1, 2, 3]
    dtype: float
  "track_index_to_predict": # [B]
    value: arange
    shape: [1]
    dtype: int
  "intention_points": # [B, 64, 2]
    value: ones
    shape: [1, 64, 2]
    dtype: float
